pkg/controllers/webui/resource.go

```diff
@@ -0,0 +1,93 @@
+package webui
+
+import (
+	"github.com/matrixorigin/matrixone-operator/api/core/v1alpha1"
+	"github.com/matrixorigin/matrixone-operator/pkg/controllers/common"
+	recon "github.com/matrixorigin/matrixone-operator/runtime/pkg/reconciler"
+	"github.com/openkruise/kruise-api/apps/pub"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func syncReplicas(wi *v1alpha1.WebUI, dp *appsv1.Deployment) {
+	dp.Spec.Replicas = &wi.Spec.Replicas
+}
+
+func syncPodMeta(wi *v1alpha1.WebUI, dp *appsv1.Deployment) {
+	wi.Spec.Overlay.OverlayPodMeta(&dp.Spec.Template.ObjectMeta)
+}
+
+func syncPodSpec(wi *v1alpha1.WebUI, dp *appsv1.Deployment) {
+	main := corev1.Container{
+		Image:           wi.Spec.Image,
+		ImagePullPolicy: wi.Spec.Overlay.ImagePullPolicy,
+		Name:            v1alpha1.ContainerMain,
+		Resources:       wi.Spec.Resources,
+		LivenessProbe:   wi.Spec.Overlay.LivenessProbe,
+		ReadinessProbe:  wi.Spec.Overlay.ReadinessProbe,
+		StartupProbe:    wi.Spec.Overlay.StartupProbe,
+		Lifecycle:       wi.Spec.Overlay.Lifecycle,
+		SecurityContext: wi.Spec.Overlay.ContainerSecurityContext,
+	}
```

Comment:
I suggest build the container and then apply the overlay, so that we don't have to update multiple places when we add an new overlay field